src/main.rs

```diff
@@ -250,23 +263,24 @@ fn main_try() -> Result<(), failure::Error> {
 
     let list = Probe::list_all();
 
-    let device = match opt.n {
-        Some(index) => list.get(index).ok_or_else(|| {
-            format_err!("Unable to open probe with index {}: Probe not found", index)
-        })?,
+    // If we got a probe selector as an argument, open the probe matching the selector if possible.
+    let mut probe = match opt.probe_selector {
+        Some(selector) => Probe::open(selector)?,
         None => {
             // Only automatically select a probe if there is only
             // a single probe detected.
             if list.len() > 1 {
-                return Err(format_err!("More than a single probe detected. Use the --probe-index argument to select which probe to use."));
+                return Err(format_err!("More than a single probe detected. Use the --probe-selector argument to select which probe to use."));
```

Comment:
I would suggest naming this just `probe`. probe-selector sounds a bit weird to me.